src/mame/drivers/blit.cpp

```diff
@@ -0,0 +1,312 @@
+	}
+	else
+	{
+		fg = 1; bg = 0;
+	}
+
+	for (y = 0; y < 1024; y++)
+	{
+		uint16_t *p = &bitmap.pix(y);
+
+		for (x = 0; x < 800 / 8; x += 2)
+		{
+			gfx = m_videoram[x + y * 100 + 1];
+
+			for (z = 7; z >= 0; z--)
+			{
+				*p++ = BIT(gfx, z) ? fg : bg;
+			}
+
+			gfx = m_videoram[x + y * 100];
+
+			for (z = 7; z >= 0; z--)
+			{
+				*p++ = BIT(gfx, z) ? fg : bg;
+			}
+		}
+	}
+	return 0;
+}
+
+
+void blit_state::machine_start()
+{
+	m_videoram = (uint8_t *)m_maincpu->space(AS_PROGRAM).get_read_ptr(m_videostart);
+	assert(m_videoram != NULL);
+
+	save_pointer(NAME(m_sysram), sizeof(m_sysram));
+}
+
+void blit_state::machine_reset()
+{
+	*m_misccr = 0;
+
+	m_sysram_on = false;
+	memcpy(m_p_ram.target(), m_sysrom, 8);
+
+	m_maincpu->reset();
+}
+
+void blit_state::blit(machine_config &config)
+{
+	M68000(config, m_maincpu, XTAL(33'000'000) / 4); // maybe XTAL(32'760'000)
+	m_maincpu->set_addrmap(AS_PROGRAM, &blit_state::blit_mem);
+
+	SCREEN(config, m_screen, SCREEN_TYPE_LCD, rgb_t::white());
+	m_screen->set_refresh_hz(60);
+	m_screen->set_size(800, 1024);
+	m_screen->set_visarea(0, 800-1, 0, 1024-1);
+	m_screen->set_screen_update(FUNC(blit_state::screen_update));
+	m_screen->screen_vblank().set_inputline(M68K_TAG, M68K_IRQ_1);
+	m_screen->set_palette("palette");
+
+	PALETTE(config, "palette", palette_device::MONOCHROME);
+
+	CLOCK(config, "system_clock", 19200 * 16).signal_handler().set(FUNC(blit_state::system_clock_write));
+
+	ACIA6850(config, m_acia0, 0);
+	m_acia0->txd_handler().set(RS232_H_TAG, FUNC(rs232_port_device::write_txd));
+	m_acia0->rts_handler().set(RS232_H_TAG, FUNC(rs232_port_device::write_rts));
+	m_acia0->irq_handler().set_inputline(M68K_TAG, M68K_IRQ_5);
+
+	rs232_port_device &rs232h(RS232_PORT(config, RS232_H_TAG, default_rs232_devices, "null_modem"));
+	rs232h.rxd_handler().set(m_acia0, FUNC(acia6850_device::write_rxd));
+	rs232h.dcd_handler().set(m_acia0, FUNC(acia6850_device::write_dcd));
+	rs232h.cts_handler().set(m_acia0, FUNC(acia6850_device::write_cts));
+	rs232h.set_option_device_input_defaults("null_modem", DEVICE_INPUT_DEFAULTS_NAME(host_rs232_defaults));
+
+	ACIA6850(config, m_acia1, 0);
+	m_acia1->txd_handler().set(RS232_K_TAG, FUNC(rs232_port_device::write_txd));
+	m_acia1->rts_handler().set(RS232_K_TAG, FUNC(rs232_port_device::write_rts));
+	m_acia1->irq_handler().set_inputline(M68K_TAG, M68K_IRQ_2);
+
+	rs232_port_device &rs232k(RS232_PORT(config, RS232_K_TAG, default_rs232_devices, "keyboard"));
+	rs232k.rxd_handler().set(m_acia1, FUNC(acia6850_device::write_rxd));
+	rs232k.dcd_handler().set(m_acia1, FUNC(acia6850_device::write_dcd));
+	rs232k.cts_handler().set(m_acia1, FUNC(acia6850_device::write_cts));
+	rs232k.set_option_device_input_defaults("keyboard", DEVICE_INPUT_DEFAULTS_NAME(kbd_rs232_defaults));
+
+	RAM(config, RAM_TAG).set_default_size("256K");
+}
+
+ROM_START( blit )
+	ROM_REGION16_BE(0x6000, M68K_TAG, 0)
+	ROM_DEFAULT_BIOS("blit")
+
+	ROM_SYSTEM_BIOS(0, "blit", "blit ROM")
```

Comment:
Is there known to be different ROM for this hardware?

Reply:
Yes -- they are referred to in termcap entry